### PR TITLE
Enabling translations

### PR DESCRIPTION
**Pull request by tclaus**
Adding translation macros (thanks @hfsfox )
Enabling translation in the make files (WIP)
At last a $: make catkeys should run and create the locales file.

### CODE REVIEW

File: source/gui/chartView/ChartTimeRangeBar.cpp [hunk 1]
```diff
@@ -19,18 +24,18 @@ ChartTimeRangeBar::~ChartTimeRangeBar() {
 
 void
 ChartTimeRangeBar::InitTabs() {
-    BButton *oneDayButton = createButton("1 D", new BMessage(TimeRange::M_DAY));
+    BButton *oneDayButton = createButton(B_TRANSLATE("1 D"), new BMessage(TimeRange::M_DAY));
```
Comment (humdingerb) [inline source/gui/chartView/ChartTimeRangeBar.cpp:27]: When using abbreviations, or any non-obvious strings in the GUI, it's a good idea to use B_TRANSLATE_COMMENT to help the translators.

```suggestion
    BButton *oneDayButton = createButton(B_TRANSLATE_COMMENT("1 D", "Abbreviation for 'one day', short as possible."),
new BMessage(TimeRange::M_DAY));
```

File: source/gui/chartView/ChartView.cpp [hunk 2]
```diff
@@ -2,14 +2,18 @@
...
 #include "Portfolio.h"
 #include "TimeRange.h"
 #include "ApiBuilder.h"
 
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "ChartView"
+
 ChartView::ChartView() :
-        BView(BRect(), "chartView", B_FOLLOW_ALL, B_WILL_DRAW),
+        BView(BRect(), B_TRANSLATE("chartView"), B_FOLLOW_ALL, B_WILL_DRAW),
```
Comment (humdingerb) [inline source/gui/chartView/ChartView.cpp:16]: No need to translate internal view names. Could even backfire if people do some hey-scripting.

File: source/gui/chartView/DetailsDataList.cpp [hunk 3]
```diff
@@ -2,14 +2,18 @@
...
 #include "DetailsDataList.h"
 #include "Portfolio.h"
 #include "QuoteFormatter.h"
 
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "DetailsDataList"
+
 DetailsDataList::DetailsDataList() :
-        BView(BRect(), "detailsDataList", B_FOLLOW_ALL, B_WILL_DRAW) {
+        BView(BRect(), B_TRANSLATE("detailsDataList"), B_FOLLOW_ALL, B_WILL_DRAW) {
```
Comment (humdingerb) [inline source/gui/chartView/DetailsDataList.cpp:16]: Again internal name and should use layout-aware c'tor.

File: source/gui/chartView/DetailsDataList.cpp [hunk 4]
```diff
@@ -58,39 +62,39 @@ void DetailsDataList::UpdateStatus() {
...
     fLowValueStringView = new BStringView("strLowValue", "-");
     FormatValueLabel(fLowValueStringView);
 
-    fVolumeStringView = new BStringView("strVolume", "Volume");
+    fVolumeStringView = new BStringView("strVolume", B_TRANSLATE("Volume"));
     fVolumeValueStringView = new BStringView("strVolumeValue", "-");
     FormatValueLabel(fVolumeValueStringView);
 
-    fPeStringView = new BStringView("strPe", "PE");
+    fPeStringView = new BStringView("strPe", B_TRANSLATE("PE"));
```
Comment (humdingerb) [inline source/gui/chartView/DetailsDataList.cpp:81]: This and the below could also do with a B_TRANSLATE_COMMENT.

File: source/gui/chartView/DetailsHeadline.cpp [hunk 5]
```diff
@@ -5,11 +5,15 @@
...
+#include <Catalog.h>
 #include <LayoutBuilder.h>
 #include <StringView.h>
 
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "DetailsHeadline"
+
 DetailsHeadline::DetailsHeadline() :
-        BView(BRect(), "detailsHeadView", B_FOLLOW_ALL, B_WILL_DRAW) {
+        BView(BRect(), B_TRANSLATE("detailsHeadView"), B_FOLLOW_ALL, B_WILL_DRAW) {
```
Comment (humdingerb) [inline source/gui/chartView/DetailsHeadline.cpp:16]: Again internal name and should use layout-aware c'tor.

File: source/gui/chartView/DetailsHeadline.cpp [hunk 6]
```diff
@@ -82,25 +86,25 @@ void DetailsHeadline::UpdateShortSymbolName() { fSymbolShortNameLabel->SetText(f
...
 DetailsHeadline::CreateSymbolFullNameLabel() {
-    fSymbolFullNameLabel = new BStringView("symbolFullNameLabel", "Symbol");
+    fSymbolFullNameLabel = new BStringView("symbolFullNameLabel", B_TRANSLATE("Symbol"));
     fSymbolFullNameLabel->ResizeBy(0, 50);
 }
 
 void
 DetailsHeadline::CreateSymbolPriceLabel() {
-    fSymbolPriceLabel = new BStringView("symbolPriceLabel", "0,00€");
+    fSymbolPriceLabel = new BStringView("symbolPriceLabel", B_TRANSLATE("0,00€"));
```
Comment (humdingerb) [inline source/gui/chartView/DetailsHeadline.cpp:95]: Maybe in another commit, you may want to check out locale/Format.h and support/String.h's SetToFormat() to get the current locale's currency.

File: source/gui/chartView/DetailsHeadline.cpp [hunk 7]
```diff
@@ -82,25 +86,25 @@ void DetailsHeadline::UpdateShortSymbolName() { fSymbolShortNameLabel->SetText(f
...
     BFont font(be_bold_font);
     font.SetFace(B_BOLD_FACE);
     fSymbolPriceLabel->SetFont(&font);
 }
 
 void DetailsHeadline::CreateSymbolChangeLabel() {
-    fSymbolChangeLabel = new BStringView("symbolChangeLabel", "0%");
+    fSymbolChangeLabel = new BStringView("symbolChangeLabel", B_TRANSLATE("0%"));
```
Comment (humdingerb) [inline source/gui/chartView/DetailsHeadline.cpp:102]: There's locale/BNumberFormat.h to deal with percentages.

File: source/gui/chartView/DetailsHeadline.cpp [hunk 8]
```diff
@@ -82,25 +86,25 @@ void DetailsHeadline::UpdateShortSymbolName() { fSymbolShortNameLabel->SetText(f
...
 }
 
 void
 DetailsHeadline::CreateSymbolShortNameLabel() {
-    fSymbolShortNameLabel = new BStringView("shortSymbolName", "SYM");
+    fSymbolShortNameLabel = new BStringView("shortSymbolName", B_TRANSLATE("SYM"));
```
Comment (humdingerb) [inline source/gui/chartView/DetailsHeadline.cpp:107]: Worth a B_TRANSLATE_COMMENT ?

File: source/gui/chartView/DetailsView.cpp [hunk 9]
```diff
@@ -3,11 +3,15 @@
...
+#include <Catalog.h>
 #include <LayoutBuilder.h>
 #include <View.h>
 
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "DetailsView"
+
 DetailsView::DetailsView()
-        : BView(BRect(), "chartView", B_FOLLOW_ALL, B_WILL_DRAW) {
+        : BView(BRect(), B_TRANSLATE("chartView"), B_FOLLOW_ALL, B_WILL_DRAW) {
```
Comment (humdingerb) [inline source/gui/chartView/DetailsView.cpp:14]: Again internal name and should use layout-aware c'tor.

File: source/gui/chartView/DetailsHeadline.cpp (hunk 6 above)
Comment (tclaus) [reply source/gui/chartView/DetailsHeadline.cpp:95]: I'll remove the currency for now.

File: source/gui/chartView/DetailsHeadline.cpp (hunk 7 above)
Comment (tclaus) [reply source/gui/chartView/DetailsHeadline.cpp:102]: Thanks for this info, here this is only a placeholder at app's start. The real value will be calculated / inserted by the API. So it is not needed to translate '0%' in any way I think.

**Comment by tclaus** [comment]
@humdingerb Thanks for your review - the ctror's I will check or change in another PR on short notice.

File: source/gui/chartView/DetailsHeadline.cpp (hunk 6 above)
Comment (humdingerb) [reply source/gui/chartView/DetailsHeadline.cpp]: I guess the app will be using the currency of the view stock anyway. Still, decimal and 1,000 separators ("." or ",") is still something locale dependent.

File: source/gui/chartView/ChartTimeRangeBar.cpp [hunk 10]
```diff
@@ -24,18 +24,26 @@ ChartTimeRangeBar::~ChartTimeRangeBar() {
 
 void
 ChartTimeRangeBar::InitTabs() {
-    BButton *oneDayButton = createButton(B_TRANSLATE("1 D"), new BMessage(TimeRange::M_DAY));
+    BButton *oneDayButton = createButton(B_TRANSLATE_COMMENT("1 D", "Abbreviation for 'one day', short as possible"),
```
Comment (humdingerb) [inline source/gui/chartView/ChartTimeRangeBar.cpp:27]: JFYI, with "short as possible", I didn't mean that this is the shortest intervall the app can show, but that the translator should make sure they abbreviate as small as possible (if the GUI at that location demands to use the smallest possible space). :)

Comment (tclaus) [reply source/gui/chartView/ChartTimeRangeBar.cpp:27]: Ah.. thanks. Both point of views certainly help.

**Comment by augiedoggie** [comment]
Does the `make catkeys` step work for you?
I wasn't able to get it to work.  In order for me to build both the app and catkeys I need to add a `build/` prefix to the two generated source files in the CMakeLists.txt file.   There is an issue with the Haiku cmake module that I'll have to fix so that this isn't necessary.

I needed to add a `build/` prefix here:
https://github.com/tclaus/stocks/blob/enabling_translations/CMakeLists.txt#L157
and
https://github.com/tclaus/stocks/blob/enabling_translations/CMakeLists.txt#L181

**Pull request merged**